prog/dftb+/api/mm/mmapi.F90

```diff
@@ -467,6 +473,64 @@ subroutine convertAtomTypesToSpecies(typeNumbers, species, speciesNames, typeNam
 
   end subroutine convertAtomTypesToSpecies
 
+  !TODO(Alex) Consider whether this should be a member function or free function
+  !> Checks speciesNames has not changed between calls to DFTB+
+  subroutine TDftbPlus_checkSpeciesNames(this, inputSpeciesNames)
+    !TODO(Alex) Establish how to set len of inputSpeciesNames
+    !use dftbp_accuracy,    only: mc
+    use dftbp_message, only: error
+    class(TDftbPlus),               intent(inout) :: this
+    character(len=3),  allocatable, intent(in)    :: inputSpeciesNames(:)
```

Comment:
Pass mc to set length?

Reply:
Use `character(*)`. Is the `allocatable` attribute necessary?